lib/bigquery/index.js

```diff
@@ -96,15 +96,10 @@ var SCOPES = ['https://www.googleapis.com/auth/bigquery'];
  */
 function BigQuery(options) {
   if (!(this instanceof BigQuery)) {
+    options = util.normalizeArguments(this, options);
```

Comment:
Instead of exposing factory patterns, we just expose the constructors now and the user treats them like a factory. Ultimately what ends up happening is the constructor gets called, sees that it's not instantiated and returns a new instance of itself (aka calls itself a second time)